src/Microsoft.SqlTools.ServiceLayer/Utility/AccessTokenProvider.cs

```diff
@@ -0,0 +1,24 @@
+﻿using Microsoft.SqlServer.Dac;
+using System;
+
+namespace Microsoft.SqlTools.ServiceLayer.Utility
+{
+    class AccessTokenProvider : IUniversalAuthProvider
+    {
+        private string _accessToken;
+
+        public AccessTokenProvider(string accessToken)
+        {
+            if (string.IsNullOrEmpty(accessToken))
+            {
+                throw new ArgumentNullException("accessToken");
+            }
+
+            _accessToken = accessToken;
+        }
+
+        public bool IsTokenExpired() { return false; }
```

Comment:
never expires? or we don't care in current case?

Reply:
Yeah I don't think we care since the token is passed from ADS. This is the same implementation that the SqlPackageAccessTokenProvider has